pepper-apis/studybuilder-cli/tenants/register-user-in-pepper-action.js

```diff
@@ -259,13 +273,17 @@ exports.onExecutePostLogin = async (event, api) => {
       console.log('pepper params: ' + JSON.stringify(pepper_params));
       console.debug('EVENT: ' + JSON.stringify(event));
       console.debug('API::: ' + JSON.stringify(api));
-      if (event.user.app_metadata.user_guid) {
-        console.log('setting userGUID claim from user.app_metadata');
-        api.idToken.setCustomClaim(pepperUserGuidClaim, event.user.app_metadata.user_guid);
-      } else if (pepper_params.tempUserGuid) {
+
+      if (pepper_params.mode && pepper_params.mode === 'signup' && pepper_params.tempUserGuid) {
         console.log('setting userGUID claim from temp user');
         api.idToken.setCustomClaim(pepperUserGuidClaim, pepper_params.tempUserGuid);
         api.user.setAppMetadata("user_guid", pepper_params.tempUserGuid);
+        api.user.setAppMetadata("study_guid", pepper_params.studyGuid);
```

Comment:
added study-guid to user AppMetadata so that we can check during login and avoid backend call if not registered to this study